fem/bilininteg_hcurl.cpp

```diff
@@ -419,11 +431,39 @@ static void PAHcurlMassAssembleDiagonal3D(const int D1D,
 void VectorFEMassIntegrator::AssembleDiagonalPA(Vector& diag)
 {
    if (dim == 3)
-      PAHcurlMassAssembleDiagonal3D(dofs1D, quad1D, ne,
-                                    mapsO->B, mapsC->B, pa_data, diag);
+   {
+      if (fetype == mfem::FiniteElement::CURL)
+      {
+         PAHcurlMassAssembleDiagonal3D(dofs1D, quad1D, ne,
+                                       mapsO->B, mapsC->B, pa_data, diag);
+      }
+      else if (fetype == mfem::FiniteElement::DIV)
+      {
+         PAHdivMassAssembleDiagonal3D(dofs1D, quad1D, ne,
+                                      mapsO->B, mapsC->B, pa_data, diag);
+      }
+      else
+      {
+         MFEM_ABORT("Unknown kernel.");
+      }
+   }
    else
-      PAHcurlMassAssembleDiagonal2D(dofs1D, quad1D, ne,
-                                    mapsO->B, mapsC->B, pa_data, diag);
+   {
+      if (fetype == mfem::FiniteElement::CURL)
+      {
+         PAHcurlMassAssembleDiagonal2D(dofs1D, quad1D, ne,
+                                       mapsO->B, mapsC->B, pa_data, diag);
+      }
+      else if (fetype == mfem::FiniteElement::DIV)
+      {
+         PAHdivMassAssembleDiagonal2D(dofs1D, quad1D, ne,
+                                      mapsO->B, mapsC->B, pa_data, diag);
+      }
+      else
+      {
+         MFEM_ABORT("Unknown kernel.");
```

Comment:
It would make sense to have a file for VectorFE, but so far there would be a small amount of code in it. I am fine with doing it either way, but would think it is fine as is for now. @YohannDudouit @v-dobrev Does anyone else have a preference?

Reply:
Ultimately, as we add more integrators that work with both H(div) and H(curl), we will need to create a new file probably. However, I don't think we need to do it right now.